packages/flutter/lib/src/cupertino/sheet.dart

```diff
@@ -0,0 +1,499 @@
+// of the sheet in the background and comparing against the screen width on the
+// iOS simulator showing an iPhone 16 pro running iOS 16.0. The scale transition
+//will go from a default of 1.0 to 1.0 - _kSheetScaleFactor.
+const double _kSheetScaleFactor = 0.0835;
+
+final Animatable<double> _kScaleTween = Tween<double>(begin: 1.0, end: 1.0 - _kSheetScaleFactor);
+
+/// Shows a Cupertino style sheet widget that slides up from the bottom of the
+/// screen and stacks the previous route behind the new sheet.
+///
+/// This is a convenience method for displaying [CupertinoSheetRoute] for common,
+/// straightforward use cases. The Widget returned from `pageBuilder` will be
+/// used to display the content on the [CupertinoSheetRoute].
+///
+/// `useNestedNavigation` allows new routes to be pushed inside of a [CupertinoSheetRoute]
+/// by adding a new [Navigator] inside of the [CupertinoSheetRoute].
+///
+/// When `useNestedNavigation` is set to `true`, any route pushed to the stack
+/// from within the context of the [CupertinoSheetRoute] will display within that
+/// sheet. System back gestures and programatic pops on the initial route in a
+/// sheet will also be intercepted to pop the whole [CupertinoSheetRoute]. If
+/// a custom [Navigator] setup is needed, like for example to enable named routes
+/// or the pages API, then it is recommended to directly push a [CupertinoSheetRoute]
+/// to the stack with whatever configuration needed. See [CupertinoSheetRoute] for
+/// an example that manually sets up nested navigation.
+///
+/// The whole sheet can be popped at once by either dragging down on the sheet,
+/// or calling [CupertinoSheetRoute.popSheet].
+///
+/// iOS sheet widgets are generally designed to be tightly coupled to the context
+/// of the widget that opened the sheet. As such, it is not recommended to push
+/// a non-sheet route that covers the sheet without first popping the sheet. If
+/// necessary however, it can be done by pushing to the root Navigator.
+///
+/// If `useNestedNavigation` is `false` (the default), then a [CupertinoSheetRoute]
+/// will be shown with no [Navigator] widget. Multiple calls to `showCupertinoSheet`
+/// can still be made to show multiple stacked sheets, if desired.
+///
+/// `showCupertinoSheet` always pushes the [CupertinoSheetRoute] to the root
+/// [Navigator]. This is to ensure the previous route animates correctly.
+///
+/// Returns a [Future] that resolves to the value (if any) that was passed to
+/// [Navigator.pop] when the sheet was closed.
+///
+/// {@tool dartpad}
+/// This example shows how to navigate to use [showCupertinoSheet] to display a
+/// Cupertino sheet widget with nested navigation.
+///
+/// ** See code in examples/api/lib/cupertino/sheet/cupertino_sheet.1.dart **
+/// {@end-tool}
+///
+/// See also:
+///
+///  * [CupertinoSheetRoute] the basic route version of the sheet view.
+///  * [showCupertinoDialog] which displays an iOS-styled dialog.
```

Comment:
Should we link to the Swift UI docs, or is there anywhere else in Apple's docs that describes what a sheet is? https://developer.apple.com/documentation/swiftui/view/sheet(ispresented:ondismiss:content:)